Storage/Service/EntityStorage.php

```diff
@@ -39,131 +33,60 @@ public function __construct(Registry $doctrine, bool $contextSupport)
-            return $this->repository->head($hash, $cacheContext);
-        }
-        return false;
-    }
-
-    public function getSize(string $hash, ?string $cacheContext = null): ?int
-    {
-        if (!$cacheContext || $this->contextSupport) {
-            return $this->repository->getSize($hash, $cacheContext);
-        }
-        return null;
+        return $this->repository->getSize($hash);
     }
 
 
-    /**
-     * @param string $hash
-     * @param string $filename
-     * @param string|null $cacheContext
-     * @return bool
-     */
-    public function create(string $hash, string $filename, ?string $cacheContext = null): bool
+    public function create(string $hash, string $filename): bool
     {
-        if (!$cacheContext || $this->contextSupport) {
-            $entity = $this->createEntity($hash, $cacheContext);
-
-            $entity->setSize(filesize($filename));
-            $entity->setContents(file_get_contents($filename));
-            $entity->setLastUpdateDate(filemtime($filename));
-            $entity->setConfirmed(true);
-
-            $this->manager->persist($entity);
-            $this->manager->flush();
+        $entity = $this->createEntity($hash);
 
-            return true;
-        }
-        return false;
+        $entity->setSize(\filesize($filename));
+        $entity->setContents(\file_get_contents($filename));
+        $entity->setConfirmed(true);
+        $this->manager->persist($entity);
+        $this->manager->flush();
     }
 
-    private function createEntity($hash, $cacheContext = false)
+    private function createEntity(string $hash)
     {
         /**@var AssetStorage $entity */
-        $entity = $this->repository->findByHash($hash, $cacheContext);
+        $entity = $this->repository->findByHash($hash);
         if (!$entity) {
             $entity = new AssetStorage();
             $entity->setHash($hash);
-            $entity->setContext($cacheContext ? $cacheContext : null);
         }
         return $entity;
     }
 
-    /**
-     * @param string $hash
-     * @param string|null $cacheContext
-     * @param bool $confirmed
-     * @return resource|null
-     */
-    public function read(string $hash, ?string $cacheContext = null, bool $confirmed = true)
+    public function read(string $hash, bool $confirmed = true): StreamInterface
     {
-        if (!$cacheContext || $this->contextSupport) {
-            /**@var AssetStorage $entity */
-            $entity = $this->repository->findByHash($hash, $cacheContext, $confirmed);
-            if ($entity) {
-                $contents = $entity->getContents();
-
-                if (is_resource($contents)) {
-                    return $contents;
-                }
-                $resource = fopen('php://memory', 'w+');
-                if ($resource === false) {
-                    return null;
-                }
-                fwrite($resource, $contents);
+        /**@var AssetStorage $entity */
```

Comment:
space between ** and @ symbol!
```suggestion
        /** @var AssetStorage $entity */
```

Reply:
aed90296d52b5cecafaafe2a1f3fea9cccd1b24f